Sources/ProjectSpec/SpecFile.swift

```diff
@@ -60,40 +60,41 @@ public struct SpecFile {
         let basePath = include.relativePaths ? (basePath + relativePath) : (basePath + relativePath + include.path.parent())
         let relativePath = include.relativePaths ? include.path.parent() : Path()
 
-        try self.init(filename: include.path.lastComponent, basePath: basePath, relativePath: relativePath)
+        try self.init(filename: include.path.lastComponent, basePath: basePath, relativePath: relativePath, shouldIgnoreNotFound: include.optional)
     }
 
-    private init(filename: String, basePath: Path, relativePath: Path = "") throws {
+    private init(filename: String, basePath: Path, relativePath: Path = "", shouldIgnoreNotFound: Bool = false) throws {
         let path = basePath + relativePath + filename
-        let jsonDictionary = try SpecFile.loadDictionary(path: path)
+        let jsonDictionary = try SpecFile.loadDictionary(path: path, shouldIgnoreNotFound: shouldIgnoreNotFound)
 
         let includes = Include.parse(json: jsonDictionary["include"])
-        let subSpecs: [SpecFile] = try includes.compactMap { include in
-            do {
-                return try SpecFile(include: include, basePath: basePath, relativePath: relativePath)
-            } catch {
-                if include.optional {
-                    return nil
-                } else {
-                    throw error
-                }
-            }
+        let subSpecs: [SpecFile] = try includes.map { include in
+            try SpecFile(include: include, basePath: basePath, relativePath: relativePath)
         }
 
         self.init(filename: filename, jsonDictionary: jsonDictionary, basePath: basePath, relativePath: relativePath, subSpecs: subSpecs)
     }
 
-    static func loadDictionary(path: Path) throws -> JSONDictionary {
+    static func loadDictionary(path: Path, shouldIgnoreNotFound: Bool) throws -> JSONDictionary {
+        let data: Data
+        do {
+            data = try path.read()
+        } catch {
+            if shouldIgnoreNotFound {
```

Comment:
I was thinking throwing a `GenerationError.includeNotFound`, which you can then catch at the top level, and it it's not optional then rethrow